web/src/components/query-browser.tsx

```diff
@@ -780,7 +780,7 @@ const QueryBrowser_: React.FC<QueryBrowserProps> = ({
                 {
                   endpoint: PrometheusEndpoint.QUERY_RANGE,
                   endTime: timeRange.endTime,
-                  namespace: perspective === 'dev' ? activeNamespace : namespace,
+                  namespace: perspective === 'dev' ? activeNamespace : '',
```

Comment:
Maybe something to check on a follow up PR. But do we need then the `namespace` on this component? It seems is only used to trigger new states on change.

Reply:
We need to trigger new states on change due to the usage in the dev perspective. If the user has queried for something like `cpu_usage` and they use the namespace selector to change namespace they should see the data from the new namespace.

Reply:
sure, but then could use the `activeNamespace` change as trigger?

Reply:
Yes very likely. I'll test out that change in a follow up PR

Reply:
Follow up PR here: #296. I tested the changes in the developer perspective and the page is able to show the new data when swapping namespace